src/lib/config_file.c

```diff
@@ -109,3 +110,28 @@ bool config_parse(const char *filename, parse_config_fn_t parse_config_fn, void
 	fclose(config_file);
 	return ret;
 }
+
+int config_write_section(FILE *file, const char *section)
+{
+	int ret = 0;
+
+	if (!file) {
+		return -EINVAL;
+	}
+	ret = fprintf(file, "[%s]\n", section);
```

Comment:
I think we should leave an empty line before and after a section so it is better formatted:
```

[SECTION1]

url=something
version=0

[SECTION2]

usr=something/else
version=0

Reply:
I personally don't like the idea on using the same separator from one section to the other and the items inside a section. This for example looks weird to me:

```
[section1]

key=value

[section2]

key=value
```

This isn't clear that the `key=value` pair is inside a section. Other problem is that it's hard not to add an empty line break before the beginning of the file, what is also weird to me. As this file isn't going to be necessarily read.

I like the style used by gitconfig, that is very easy to read, but I'm not sure if our parser supports that:

```
[section1]
    key=value
[section2]
    key=value
```